homeassistant/components/tplink/switch.py

```diff
@@ -55,20 +84,16 @@ def __init__(
         parent: Device | None = None,
     ) -> None:
         """Initialize the switch."""
+        description = _description_for_feature(
+            TPLinkSwitchEntityDescription,
+            feature,
+            SWITCH_DESCRIPTIONS_MAP,
+            child_alias=device.alias if parent else None,
```

Comment:
Would it make sense to pass the device as-is and do the handling inside `_description_for_feature`?

Reply:
Alternatively, we could intern `_description_for_feature` to avoid passing the information around?

Reply:
We can't pass the device as `_parent` is a private attribute of `device`. The `description_for_feature` is moved to a classmethod in https://github.com/home-assistant/core/pull/120064